lib/insights.ts

```diff
@@ -141,9 +141,10 @@ class AlgoliaAnalytics {
     this.viewedObjectIDs = viewedObjectIDs.bind(this);
     this.viewedFilters = viewedFilters.bind(this);
 
-    this.setUserToken(this.ANONYMOUS_USER_TOKEN);
     // Process queue upon script execution
-    this.processQueue(window);
+    if (__BROWSER_BUILD__ && typeof window !== "undefined") {
+      this.processQueue(window);
```

Comment:
I think that with different entry point for each environment would simplify the logic. We could hook the requester at that point rather than branching inside the code. We could also drop the `__BROWSER_BUILD__` variable. Only the entry point is env specific. The rest of the codebase should be env agnostic (if possible otherwise we keep some feature detection logic like cookies).

Reply:
Now at the entry, it's decided whether or not to call `processQueue`. https://github.com/algolia/search-insights.js/pull/141/commits/baa224fd2f55d5a9e70e158a35eb1a0be1f3fced